std/math/pow.zig

```diff
@@ -26,145 +26,175 @@ const std = @import("../index.zig");
 const math = std.math;
 const assert = std.debug.assert;
 
-// This implementation is taken from the go stlib, musl is a bit more complex.
-pub fn pow(comptime T: type, x: T, y: T) T {
-    if (T != f32 and T != f64) {
-        @compileError("pow not implemented for " ++ @typeName(T));
-    }
+pub fn pow(comptime T: type, x: T, y: T) (error{Overflow}!T) {
```

Comment:
Maybe this is why you closed the issue, but since the float versions of this can't overflow, float pow and int pow should probably be separate functions. I'm also OK with making this pow function undefined behavior on overflow, and adding integer support. The doc comments of the function should indicate that, for integers, it may be better to call `math.safePow` or whatever we call it, so that overflow is an error. Also to be clear, when I say "make it undefined behavior" I mean that overflow should hit an `unreachable` (or otherwise trip integer overflow detection). This will make it undefined behavior in release fast mode, but a safety panic in debug mode.

Reply:
opening another pull request because github won't let me reopen.